test/nodejs-bundle/circuit/dial-over-any-relay.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-env mocha */
+'use strict'
+
+const TCP = require('libp2p-tcp')
+const WS = require('libp2p-websockets')
+const multiplex = require('libp2p-multiplex')
+
+const waterfall = require('async/waterfall')
+const utils = require('./helpers/utils')
+
+const chai = require('chai')
+chai.use(require('dirty-chai'))
+
+const expect = chai.expect
+
+describe('test connecting over any relay', function () {
+  this.timeout(500000)
+
+  let portBase = 9010 // TODO: randomize or mock sockets
+  let testNodes
+
+  function setUpNodes (muxer, done) {
+    utils.createNodes(
+      {
+        relayNode: {
+          transports: [new TCP(), new WS()],
+          muxer: muxer,
+          addrs: [
+            `/ip4/0.0.0.0/tcp/${portBase++}`,
+            `/ip4/0.0.0.0/tcp/${portBase++}/ws`
+          ],
+          isCrypto: true,
+          config: {
+            relay: {
+              circuit: {
+                enabled: true
+              }
+            }
+          }
+        },
+        nodeA: {
+          transports: [new TCP()],
+          isCrypto: true,
+          muxer: muxer,
+          addrs: [
+            `/ip4/0.0.0.0/tcp/${portBase++}`
+          ]
+        },
+        nodeB: {
+          transports: [new WS()],
+          isCrypto: true,
+          muxer: muxer,
+          addrs: [
+            `/ip4/0.0.0.0/tcp/${portBase++}/ws`
+          ]
+        }
+      },
+      (err, nodes) => {
+        if (err) {
+          return done(err)
+        }
+
+        testNodes = nodes
+        done()
+      })
+  }
```

Comment:
Use createNode https://github.com/libp2p/js-libp2p/blob/master/test/nodejs-bundle/utils.js#L12-L34, this is adding a lot of code duplication.

Reply:
Why does this need to be so complicated? https://github.com/libp2p/js-libp2p/blob/master/test/nodejs-bundle/tcp%2Bwebsockets%2Bwebrtc-star.js boots a lot of nodes without requiring so much customization and. code duplication

- Why do you need to set up the transports explicitly? Libp2p only boots transports it has listeners to
- Why do you need to setup muxer and isCrypto?